### PR TITLE
Fix Sphinxsetup.sh script for Ubuntu 18.04 and 20.04

### PR DESCRIPTION
**Pull request by TunaLobster**
Hit an issue while trying to use the script on bionic WSL 1. I've narrowed the scope of when the script can be used.

### CODE REVIEW

**Comment by TunaLobster** [comment]
Something doesn't seem correct with how I've used git on this one. Any helpful hints?

**Comment by Hwurzburg** [comment]
not sure why you are trying to add an existing document...need to rebase perhaps?

**Comment by TunaLobster** [comment]
I think it has to do with the file being in the .gitignore. I have the file locally when I rebased this morning.

**Comment by TunaLobster** [comment]
Somehow my rebase didn't move the files from /dev to /common. Let's try this again.

**Comment by TunaLobster** [comment]
@peterbarker Could you review this change?

**Review by peterbarker — COMMENTED** [review]
Have those Sphinxsetup.sh changes actually been tested in their current form?

Note that you can bring up a VM to test this with minor modifications to the Vagrantfie.

**Comment by TunaLobster** [comment]
Whoops! That was not great. Tested on WLS 1 with bionic and Vagrant with focal64. Got the expected results. Tried to test in WSL 2 with focal, but could not get do-release-upgrade to behave.

File: Sphinxsetup.sh [hunk 1]
```diff
@@ -7,27 +7,41 @@ if [ "$UID" -gt 0 ]; then
      exit 1
 fi
 
-DISTRIBUTION_CODENAME=$(lsb_release -i -s)
-if [ ${DISTRIBUTION_CODENAME} == 'Ubuntu' ]; then
-  add-apt-repository universe
+DISTRIBUTION_ID=$(lsb_release -i -s)
+if [ ${DISTRIBUTION_ID} == 'Ubuntu' ]; then
```
Comment (khancyr) [inline Sphinxsetup.sh:11]: This isn't good as it is working perfectly on debian

File: Sphinxsetup.sh [hunk 2]
```diff
@@ -7,27 +7,41 @@ if [ "$UID" -gt 0 ]; then
...
 # Get pip through the official website to get the lastest release
 curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py && python3 get-pip.py && rm -f get-pip.py
 
 # Install sphinx
-pip install -U sphinx==1.8.3 || pip3 install -U sphinx==1.8.3
+pip install --upgrade sphinx==1.8.3 || pip3 install --upgrade sphinx==1.8.3
 
 # Install sphinx theme from ArduPilot repository
-pip install git+https://github.com/ArduPilot/sphinx_rtd_theme.git -UI ||  pip3 install git+https://github.com/ArduPilot/sphinx_rtd_theme.git -UI
+pip install git+https://github.com/ArduPilot/sphinx_rtd_theme.git --upgrade ||  pip3 install git+https://github.com/ArduPilot/sphinx_rtd_theme.git --upgrade
```
Comment (khancyr) [inline Sphinxsetup.sh:38]: This won't do it, it got a pr that correct that

**Review by khancyr — CHANGES_REQUESTED** [review]

File: Sphinxsetup.sh (hunk 2 above)
Comment (peterbarker) [reply Sphinxsetup.sh]: This certainly fixes the problem in front of us - the `-I` was causing Sphinx to be updated to a version that isn't compatible with our configuration files.

File: Sphinxsetup.sh (hunk 1 above)
Comment (peterbarker) [reply Sphinxsetup.sh:11]: Fixed.

**Pull request merged**